models/articles.model.js

```diff
@@ -1,31 +1,44 @@
 const db = require("../db/connection");
 
 function selectArticleById(id) {
-  const query = `SELECT * FROM articles WHERE article_id=$1;`;
-  return db.query(query, [id]).then(({ rows }) => {
-    if (!rows[0]) {
+  return selectArticles(id).then((article) => {
+    if (!article[0]) {
       return Promise.reject({ status: 404, msg: "article does not exist" });
     }
-    return rows[0];
+    return article[0];
   });
```

Comment:
You are now putting a lot of weight on your selectArticles model to do the work of selectArticleById. I think this refactor isn't productive and arguably makes your MVC structure harder to navigate. I see the desire to dry up code but the amount of logic you have to put into selectArticles feels like its counteracting that DRYing up you've tried to do.  Otherwise we would just be advising you to make one model with loads of conditional logic so it can handle every db query in one function.

I would prefer you to keep the two models separate but at the end of the day it is totally up to you how to implement these endpoints as long as it passes the tests